source/postprocess/visualization.cc

```diff
@@ -97,6 +97,53 @@ namespace aspect
             return update_values;
           }
       };
+
+      /**
+       * This Postprocessor will generate the output variables of mesh velocity
+       * for when a free surface is used.
+       */
+      template <int dim>
+      class FreeSurfacePostprocessor: public DataPostprocessor< dim >, public SimulatorAccess<dim>
```

Comment:
You could make your life substantially easier if you'd derive from `DataPostprocessorVector` instead of `DataPostprocessor`. Much less boiler-plate code.